shortfin/requirements-iree-compiler.txt

```diff
@@ -1,4 +1,4 @@
-# Keep in sync with "ref: iree-" in .github/workflows/* and GIT_TAG in CMakeLists.txt
+# Keep in sync with SHORTFIN_IREE_GIT_TAG in CMakeLists.txt
 -f https://iree.dev/pip-release-links.html
 iree-base-compiler==3.1.0rc20241204
 iree-base-runtime==3.1.0rc20241204
```

Comment:
@marbre did you experiment with having the CMake file pull from this requirements file?

I guess we can also just add a comment in this other location pointing back here too:
https://github.com/nod-ai/shark-ai/blob/5f08cb2354c928476ace3cfdcda3b835dd13cd61/shortfin/CMakeLists.txt#L46-L47

Reply:
Yeah, some time ago I sent #502. Happy to provide more context (and do the actual review) later.

Reply:
We discussed this a bit offline. I think having two locations that point to each other is pretty manageable without any extra tooling support. Having five locations with only some pointing to the others is not. So for now I think keeping the requirements and CMakeLists only loosely linked via comments is sufficient.